internal/cmd/operator_uninstall.go

```diff
@@ -16,42 +19,70 @@ func newOperatorUninstallCmd(cfg *action.Configuration) *cobra.Command {
-uninstall without deleting anything. 
-The "ignore" strategy keeps the operands on cluster and deletes the subscription and the operator.
-The "delete" strategy deletes the subscription, operands, and after they have finished finalizing, the operator itself.
-
-Setting --delete-operator-groups to true will delete the operatorgroup in the provided namespace if no other active 
-subscriptions are currently in that namespace, after removing the operator. The subscription and operatorgroup will be 
-removed even if the operator is not found.`,
+		Long: `Uninstall removes the subscription, operator and optionally operands managed
+by the operator as well as the relevant operatorgroup.
+
+Warning: this command permanently deletes objects from the cluster. Running
+uninstall concurrently with other operations could result in undefined behavior.
+
+The uninstall command first checks to find the subscription associated with the
+operator. It then lists all operands found throughout the cluster for the
+operator specified if one is found. Since the scope of an operator is restricted
+by its operator group, this search will include namespace-scoped operands from
+the operator group's target namespaces and all cluster-scoped operands.
+
+The operand-deletion strategy is then considered if any operands are found
+on-cluster. One of cancel|ignore|delete. By default, the strategy is "cancel",
+which means that if any operands are found when deleting the operator abort the
+uninstall without deleting anything. The "ignore" strategy keeps the operands on
+cluster and deletes the subscription and the operator. The "delete" strategy
+deletes the subscription, operands, and after they have finished finalizing, the
+operator itself.
+
+Setting --delete-operator-groups to true will delete the operatorgroup in the
+provided namespace if no other active subscriptions are currently in that
+namespace, after removing the operator. The subscription and operatorgroup will
+be removed even if the operator is not found.
+
+There are other deletion flags for removing additional objects, such as custom
+resource definitions, operator objects, and any other objects deployed alongside
+the operator (e.g. RBAC objects for the operator). These flags are:
+
+  --delete-operator
+
+      Deletes all objects associated with the operator by looking up the
+      operator object for the operator and deleting every referenced object
+      and then deleting the operator object itself. This implies the flag
+      '--operand-strategy=delete' because it is impossible to delete CRDs
+      without also deleting instances of those CRDs.
+
+  -X, --delete-all
+
+      This is a convenience flag that is effectively equivalent to the flags
+      '--delete-operator=true --delete-operator-groups=true'. This is
+      most useful to completely undo the effect of a previous run of
+      'kubectl operator install <operator> --create-operator-group'
```

Comment:
This isn't _quite_ true in all cases.

If an operator (A) depends on another operator (B), and I install A on a clean cluster, OLM will also ensure B gets installed.

The `uninstall` command will not automatically uninstall B.

To do that, the plugin would need to (at uninstall time) know that:
a) B is a dependency of A
b) B was installed automatically by OLM (i.e. not manually by a user)
c) no other installed operators depend on B

We could do something similar to `apt-get autoremove` which is a separate command that uninstalls unused dependencies.

But for now, I'll update this description to explicitly state that dependent operators must be uninstalled separately.